include/nvexec/stream/sync_wait.cuh

```diff
@@ -131,7 +131,7 @@ namespace nvexec::STDEXEC_STREAM_DETAIL_NS {
             sync_wait_t, stdexec::set_value_t, Sender>) &&
           (!stdexec::tag_invocable<sync_wait_t, Sender>) &&
           stdexec::sender<Sender, __impl::__env> &&
-          stdexec::sender_to<Sender, receiver_t<Sender>>
+          stdexec::__receiver_from<receiver_t<Sender>, Sender>
```

Comment:
This change shouldn't be necessary. We should understand what's causing the concept check to fail, it's likely something wrong in one of the CUDA senders. Does clangd tell you the type of the sender that causes problems here?

Reply:
I'll double check which files showed this error. It definitely occurred in `let_{value,error,stopped}.cpp`, so possibly related to the `let_xxx` sender.

I tried to track down the problematic sender for a while until I saw @senior-zero made a similar change [here](https://github.com/NVIDIA/stdexec/commit/3c840461130ac07d06ad33f26297ef2545a7a761#diff-d68f80726b27026129cf1b54ab0c288ced5584d19580f58c988a86c11d8d3474R131).

Reply:
I've files this as #688